### PR TITLE
minor changes to make linting happier

### PR DESCRIPTION
**Pull request by fpliger**

### CODE REVIEW

**Comment by marimeireles** [comment]
Ok, the examples seem to be working fine to me (with excepting of the repl + to do list one, that needs https://github.com/pyscript/pyscript/pull/576 to be merged).

My linter is not happier, though, I don't know if you're getting these errors:

```bash
(!) Plugin typescript: @rollup/plugin-typescript TS2740: Type '{}' is missing the following properties from type 'PyodideInterface': globals, FS, pyodide_py, version, and 16 more.
src/components/base.ts: (9:5)

9     runtime = value;
      ~~~~~~~

(!) Plugin typescript: @rollup/plugin-typescript TS2361: The right-hand side of an 'in' expression must not be a primitive.
src/components/base.ts: (193:37)

193             if ('runPythonAsync' in value) {
                                        ~~~~~

src/components/base.ts: (237:41)

237                 if ('runPythonAsync' in value) {
                                            ~~~~~

(!) Plugin typescript: @rollup/plugin-typescript TS2322: Type 'unknown' is not assignable to type 'PyodideInterface'.
src/components/base.ts: (238:21)

238                     runtime = value;
                        ~~~~~~~

src/components/pyenv.ts: (12:5)

12     runtime = value;
       ~~~~~~~

(!) Plugin typescript: @rollup/plugin-typescript: Typescript 'sourceMap' compiler option must be set to generate source maps.
created build/pyscript.js, build/pyscript.min.js in 3.3s
```

I'll take a look in the code part now :)

File: pyscriptjs/src/components/base.ts [hunk 1]
```diff
@@ -120,30 +111,29 @@ export class BaseEvalElement extends HTMLElement {
...
             if (is_async) {
-                await pyodide.runPythonAsync(
+                <string>await runtime.runPythonAsync(
                     `output_manager.change(out="${this.outputElement.id}", err="${this.errorElement.id}", append=${this.appendOutput ? 'True' : 'False'})`,
                 );
-                output = await pyodide.runPythonAsync(source);
+                output = <string>await runtime.runPythonAsync(source);
             } else {
-                output = pyodide.runPython(
+                output = <string>runtime.runPython(
```
Comment (philippjfr) [inline pyscriptjs/src/components/base.ts:128]:
```suggestion
                output = <string>runtime.runPython(
```

Can't `runPython` return anything?

File: pyscriptjs/src/components/pyconfig.ts [hunk 2]
```diff
@@ -86,11 +85,11 @@ export class PyodideRuntime extends Object {
 
     async initialize() {
         loader?.log('Loading runtime...');
-        pyodideReadyPromise = loadInterpreter(this.src);
-        const pyodide = await pyodideReadyPromise;
+        // pyodideReadyPromise =
```
Comment (marimeireles) [inline pyscriptjs/src/components/pyconfig.ts:88]: This is just floating around?

File: pyscriptjs/src/components/pyconfig.ts [hunk 3]
```diff
@@ -86,11 +85,11 @@ export class PyodideRuntime extends Object {
...
+        const pyodide: PyodideInterface = await loadInterpreter(this.src);
         const newEnv = {
             id: 'a',
-            promise: pyodideReadyPromise,
+            // promise: pyodideReadyPromise,
```
Comment (marimeireles) [inline pyscriptjs/src/components/pyconfig.ts:92]: Should this be removed?

File: pyscriptjs/src/components/pyconfig.ts [hunk 4]
```diff
@@ -129,11 +128,9 @@ export class PyodideRuntime extends Object {
             console.log('------ loader closed ------');
         }
 
-        setTimeout(() => {
-            for (const initializer of postInitializers_) {
-                initializer();
-            }
-        }, 3000);
+        for (const initializer of postInitializers_) {
+            await initializer();
```
Comment (marimeireles) [inline pyscriptjs/src/components/pyconfig.ts:132]: Why did we have a wait 3000 and now we have an await?

File: pyscriptjs/src/stores.ts [hunk 5]
```diff
@@ -7,15 +7,12 @@ export type Initializer = () => Promise<void>;
 
 export type Environment = {
     id: string;
-    promise: Promise<PyodideInterface>;
+    // promise: Promise<PyodideInterface>;
```
Comment (marimeireles) [inline pyscriptjs/src/stores.ts:10]: Remove?

File: pyscriptjs/src/stores.ts [hunk 6]
```diff
@@ -7,15 +7,12 @@ export type Initializer = () => Promise<void>;
...
     runtime: PyodideInterface;
     state: string;
 };
 
-export const pyodideLoaded = writable({
-    loaded: false,
-    premise: null,
```
Comment (marimeireles) [inline pyscriptjs/src/stores.ts:17]: Are these default from `svelte`? Why aren't we using them anymore?

**Review by marimeireles — APPROVED** [review]
Cool!
Looks good to me.
I'm just wondering about a few things, but mostly because I don't understand them myself. As I said before, all examples seem to be working =)
Maybe there's some linting to be done, but other than that 👍

**Comment by fpliger** [comment]
Thanks for the feedback! I'm super swamped but will try to get back to this when I have some time :)

File: pyscriptjs/src/components/base.ts (hunk 1 above)
Comment (fpliger) [reply pyscriptjs/src/components/base.ts:128]: looking at [their docs](https://pyodide.org/en/stable/usage/api/js-api.html#pyodide.runPython ), it returns `any`

**Comment by antocuni** [comment]
I'm confused. This seems to break at least the `todo-pylist.html` example.

If I run `npm run build`, I see the same errors as Mariana:
```
(!) Plugin typescript: @rollup/plugin-typescript TS2740: Type '{}' is missing the following properties from type 'PyodideInterface': globals, FS, pyodide_py, version, and 16 more.
src/components/base.ts: (9:5)

9     runtime = value;
      ~~~~~~~

(!) Plugin typescript: @rollup/plugin-typescript TS2361: The right-hand side of an 'in' expression must not be a primitive.
src/components/base.ts: (193:37)

193             if ('runPythonAsync' in value) {
                                        ~~~~~

src/components/base.ts: (237:41)

237                 if ('runPythonAsync' in value) {
                                            ~~~~~

(!) Plugin typescript: @rollup/plugin-typescript TS2322: Type 'unknown' is not assignable to type 'PyodideInterface'.
src/components/base.ts: (238:21)

238                     runtime = value;
                        ~~~~~~~

src/components/pyenv.ts: (12:5)

12     runtime = value;
       ~~~~~~~

(!) Plugin typescript: @rollup/plugin-typescript: Typescript 'sourceMap' compiler option must be set to generate source maps.
```

And if I run `todo-pylist.html`, I see the following errors in the console:

```
base.ts:155 Uncaught TypeError: Cannot use 'in' operator to search for 'runPythonAsync' in undefined
    at base.ts:155:33
    at Object.subscribe (index.mjs:50:9)
    at HTMLElement.runAfterRuntimeInitialized (base.ts:154:23)
    at HTMLElement.connectedCallback (pybutton.ts:52:14)
    at main.ts:19:34
    at main.ts:38:2
(anonymous) @ base.ts:155
subscribe @ index.mjs:50
runAfterRuntimeInitialized @ base.ts:154
connectedCallback @ pybutton.ts:52
(anonymous) @ main.ts:19
(anonymous) @ main.ts:38
base.ts:155 Uncaught TypeError: Cannot use 'in' operator to search for 'runPythonAsync' in undefined
    at base.ts:155:33
    at Object.subscribe (index.mjs:50:9)
    at HTMLElement.runAfterRuntimeInitialized (base.ts:154:23)
    at HTMLElement.connectedCallback (pyinputbox.ts:33:14)
    at main.ts:21:36
    at main.ts:38:2
(anonymous) @ base.ts:155
subscribe @ index.mjs:50
runAfterRuntimeInitialized @ base.ts:154
connectedCallback @ pyinputbox.ts:33
(anonymous) @ main.ts:21
(anonymous) @ main.ts:38
base.ts:246 reading source
pyconfig.ts:17 config set!
pyconfig.ts:108 config set {autoclose_loader: true, runtimes: Array(1)}
pyconfig.ts:120 Initializing runtimes...
base.ts:188 RUNTIME READY undefined
base.ts:189 Uncaught TypeError: Cannot use 'in' operator to search for 'runPythonAsync' in undefined
    at base.ts:189:37
    at Object.subscribe (index.mjs:50:9)
    at CustomWidget.connectedCallback (base.ts:187:27)
    at createWidget (base.ts:220:38)
    at PyWidget.connectedCallback (base.ts:248:9)
```

**Pull request merged**